src/Sentry/Protocol/Device.cs

```diff
@@ -456,7 +456,17 @@ public static Device FromJson(JsonElement json)
         var bootTime = json.GetPropertyOrNull("boot_time")?.GetDateTimeOffset();
         var processorCount = json.GetPropertyOrNull("processor_count")?.GetInt32();
         var cpuDescription = json.GetPropertyOrNull("cpu_description")?.GetString();
-        var processorFrequency = json.GetPropertyOrNull("processor_frequency")?.GetInt32();
+
+        // TODO: For next major: Remove this and change ProcessorFrequency from int to float
+        // The Java SDK reports the processorFrequency as `double`
+        // Java https://github.com/getsentry/sentry-java/blob/9762f09afa51944b40a9b77e116a55e54636e6c5/sentry/src/main/java/io/sentry/protocol/Device.java#L130
+        int? processorFrequency = null;
+        var processorFrequencyProperty = json.GetPropertyOrNull("processor_frequency");
```

Comment:
Should we change one of our tests as well (to make sure it works when deserialising a double)?

Reply:
AFAIK, we don't have any tests for `Device` as it's not doing anything. But the type-mismatch is obvious from looking at the Java code.

Reply:
I don't want to argue against writing tests. I've added it to https://github.com/getsentry/sentry-dotnet/issues/3542